src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -534,7 +534,7 @@ internal static void EarlyParse(string[] args)
 
                 string switchKey = switchKeyResults.SwitchKey;
 
-                if (MatchSwitch(switchKey, match: "settingsfile", smallestUnambiguousMatch: "settings"))
+                if (switchKey != null && MatchSwitch(switchKey, match: "settingsfile", smallestUnambiguousMatch: "settings"))
```

Comment:
Does it make sense to absorb more than this, like with `string.IsNullOrEmpty()` or `string.IsNullOrWhitespace()`?

Reply:
We could do that, but in the debugger, I observed it was null.  I don't imagine a case where it's string.Empty.

Reply:
Just wanted to check!

Reply:
(And save some logic if it's just whitespace)

Reply:
`switchKey != null` should be sufficient as empty string or white space string can be correctly handled by `MatchSwitch`.